Wikipedia/Code/ArticleViewController+ArticleLoad.swift

```diff
@@ -0,0 +1,232 @@
+        assert(Thread.isMainThread)
+
+        guard articleLoadWaitGroup == nil else {
+            return
+        }
+        
+        articleLoadWaitGroup = DispatchGroup()
+        articleLoadWaitGroup?.enter() // will leave on setup complete
+        articleLoadWaitGroup?.notify(queue: DispatchQueue.main) { [weak self] in
+            
+            guard let self = self else {
+                return
+            }
+            
+            self.articleAsLivingDocController.articleContentFinishedLoading()
+            
+            if altTextExperimentViewModel != nil {
+                self.setupForAltTextExperiment()
+            } else {
+                self.setupFooter()
+            }
+            
+            self.shareIfNecessary()
+            self.restoreScrollStateIfNecessary()
+            self.articleLoadWaitGroup = nil
+        }
+    }
+
+    private func setupForAltTextExperiment() {
+
+        guard let altTextExperimentViewModel,
+         altTextBottomSheetViewModel != nil else {
+            return
+        }
+        
+        let oldContentInset = webView.scrollView.contentInset
+        webView.scrollView.contentInset = UIEdgeInsets(top: oldContentInset.top, left: oldContentInset.left, bottom: view.bounds.height * 0.65, right: oldContentInset.right)
+        messagingController.hideEditPencils()
+        messagingController.scrollToNewImage(filename: altTextExperimentViewModel.filename)
+        
+        presentAltTextModalSheet()
+    }
+    
+    func presentAltTextModalSheet() {
+        
+        guard altTextExperimentViewModel != nil,
+         let altTextBottomSheetViewModel else {
+            return
+        }
+
+        let bottomSheetViewController = WMFAltTextExperimentModalSheetViewController(viewModel: altTextBottomSheetViewModel, delegate: self, loggingDelegate: self)
+        
+        if #available(iOS 16.0, *) {
+            if let sheet = bottomSheetViewController.sheetPresentationController {
+                sheet.delegate = self
+                let customSmallId = UISheetPresentationController.Detent.Identifier("customSmall")
+                let customSmallDetent = UISheetPresentationController.Detent.custom(identifier: customSmallId) { context in
+                    return 44
+                }
+                sheet.detents = [customSmallDetent, .medium(), .large()]
+                sheet.selectedDetentIdentifier = .medium
+                sheet.largestUndimmedDetentIdentifier = .medium
+                sheet.prefersGrabberVisible = true
+            }
+            bottomSheetViewController.isModalInPresentation = true
+            self.altTextBottomSheetViewController = bottomSheetViewController
+            
+            present(bottomSheetViewController, animated: true) { [weak self] in
+                self?.presentAltTextTooltipsIfNecessary(force: false)
+            }
+        }
+    }
+    
+    internal func presentAltTextTooltipsIfNecessary(force: Bool = false) {
```

Comment:
Do we need to specify internal explicitly? As swift has all method as internal by default.